pkg/apis/build/v1alpha1/buildrun_types.go

```diff
@@ -48,7 +48,7 @@ type BuildRunSpec struct {
 
 	// State is used for canceling a buildrun (and maybe more later on).
 	// +optional
-	State BuildRunRequestedState `json:"state,omitempty"`
+	State *BuildRunRequestedState `json:"state,omitempty"`
```

Comment:
Optional string where we are not required to distinguish `nil` and `""`. See discussion above.

Reply:
I'm wondering if deciphering user intent is important here, and therefore a pointer would be justified.